cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/existingdataproc/ExistingDataprocConf.java

```diff
@@ -0,0 +1,291 @@
+    } catch (IOException e) {
+      throw new IOException("Unable to get credentials from the environment. "
+                              + "Please explicitly set the account key.", e);
+    }
+  }
+
+  private String getMachineType(int cpus, int memoryGB) {
+    // TODO: there are special names for pre-defined cpu and memory
+    // for example, 4cpu 3.6gb memory is 'n1-highcpu-4', 4cpu 15gb memory is 'n1-standard-4'
+    return String.format("custom-%d-%d", cpus, memoryGB);
+  }
+
+  /**
+   * Create the conf from a property map while also performing validation.
+   *
+   * @throws IllegalArgumentException if it is an invalid config
+   */
+  static ExistingDataprocConf fromProperties(Map<String, String> properties) {
+    return create(properties);
+  }
+
+  static ExistingDataprocConf create(Map<String, String> properties) {
+    String accountKey = getString(properties, "accountKey");
+    if (accountKey == null || AUTO_DETECT.equals(accountKey)) {
+      try {
+        getComputeEngineCredentials();
+      } catch (IOException e) {
+        throw new IllegalArgumentException(e.getMessage(), e);
+      }
+    }
+    String projectId = getString(properties, PROJECT_ID_KEY);
+    if (projectId == null || AUTO_DETECT.equals(projectId)) {
+      projectId = getSystemProjectId();
+    }
+    String region = getString(properties, "region");
+    if (region == null || AUTO_DETECT.equals(region)) {
+      region = getRegionFromZone(getSystemZone());
+    }
+    String clustername = getString(properties, "clusterName");
+    if (clustername == null) {
+      throw new IllegalArgumentException("Dataproc ClusterName is mandatory.");
+    }
+
+    boolean sshMonitoringEnabled = Boolean.parseBoolean(properties.get("sshMonitoringEnabled"));
+    String user = getString(properties, "user");
+    if (sshMonitoringEnabled && Strings.isNullOrEmpty(user)) {
+      throw new IllegalArgumentException("SSH monitoring is enabled but SSH user not provided for monitoring");
+
+    }
+
+    String privateKey = getString(properties, "sshKey");
+    if (sshMonitoringEnabled && Strings.isNullOrEmpty(privateKey)) {
+      throw new IllegalArgumentException("SSH monitoring is enabled but SSH key not provided for monitoring");
+
+    }
+
+    SSHKeyPair keyPair = null;
+    if (sshMonitoringEnabled) {
+      keyPair = new SSHKeyPair(new SSHPublicKey(user, ""), () -> privateKey.getBytes(StandardCharsets.UTF_8));
+    }
+    Map<String, String> labels;
+    String labelsStr = getString(properties, "labels");
+    if (Strings.isNullOrEmpty(labelsStr)) {
+      labels = Collections.emptyMap();
+    } else {
+      labels = DataprocUtils.parseKeyValueConfig(labelsStr, ";", "\\|");
+      //Validate Label key/values to conform to Requirements
+      // https://cloud.google.com/dataproc/docs/guides/creating-managing-labels
+      if (labels.size() > 64) {
+        throw new IllegalArgumentException("Exceed Max number of labels. Only Max of 64 allowed. ");
```

Comment:
Remove extra space after the `.`

Reply:
Also, use `maximum` instead of Max in the error message.

Reply:
Done